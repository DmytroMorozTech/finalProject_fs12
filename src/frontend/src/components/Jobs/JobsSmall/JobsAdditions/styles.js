import { makeStyles } from '@material-ui/core/styles'

export default makeStyles((theme) => ({

  container: {
    '& div': {
      margin: 0
    }
  },

  menuItem: {
    minHeight: theme.spacing(7),
    '&:hover': {
      backgroundColor: 'rgba(0, 0, 0, 0.08)'
    }
  }
}))
