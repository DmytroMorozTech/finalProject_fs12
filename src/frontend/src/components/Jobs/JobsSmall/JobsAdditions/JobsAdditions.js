import styles from './styles'
import NotificationsRoundedIcon from '@material-ui/icons/NotificationsRounded'
// import LocalAtmIcon from '@material-ui/icons/LocalAtm'
import AssignmentTurnedInOutlinedIcon from '@material-ui/icons/AssignmentTurnedInOutlined'
import NoteIcon from '@material-ui/icons/Note'
import InsertDriveFileOutlinedIcon from '@material-ui/icons/InsertDriveFileOutlined'
import SettingsIcon from '@material-ui/icons/Settings'
import React from 'react'
import { MenuItem } from '@material-ui/core'
import JobsLeftItem from '../../JobsLeft/JobsLeftItem/JobsLeftItem'

function JobsAdditions () {
  const classes = styles()

  /* Links is hardcoded in items */
  const items = [
    {
      Icon: <NotificationsRoundedIcon fontSize="inherit"/>,
      title: 'Job Alerts',
      to: '#'
    },
    // {
    //   Icon: <LocalAtmIcon fontSize="inherit"/>,
    //   title: 'Salary',
    //   to: '#'
    // },
    {
      Icon: <AssignmentTurnedInOutlinedIcon fontSize="inherit"/>,
      title: 'Skill Assessments',
      to: '#'
    },
    {
      Icon: <NoteIcon fontSize="inherit"/>,
      title: 'Interview Prep',
      to: '#'
    },
    {
      Icon: <InsertDriveFileOutlinedIcon fontSize="inherit"/>,
      title: 'Resume Builder',
      to: '#'
    },
    {
      Icon: <SettingsIcon fontSize="inherit"/>,
      title: 'Application Settings',
      to: '#'
    }
  ]

  return (
    <div className={classes.container}>
      {items.map(({ Icon, ...rest }, i) => (
        <MenuItem className={classes.menuItem} key={i}>
          <JobsLeftItem key={i} Icon={Icon} {...rest}/>
        </MenuItem>
      ))}
    </div>
  )
}

export default JobsAdditions
