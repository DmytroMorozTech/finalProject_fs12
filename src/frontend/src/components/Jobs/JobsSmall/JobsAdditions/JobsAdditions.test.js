import { shallow } from 'enzyme'
import React from 'react'
import JobsAdditions from './JobsAdditions'

jest.mock('react-router-dom', () => ({
  Link: 'a'
}))

const setUp = () => shallow(<JobsAdditions/>)

describe('should render JobsAdditions component', () => {
  let component
  beforeEach(() => {
    component = setUp()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
