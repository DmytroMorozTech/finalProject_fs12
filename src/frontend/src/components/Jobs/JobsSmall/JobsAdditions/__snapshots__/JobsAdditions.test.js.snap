// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render JobsAdditions component to match snapshot 1`] = `
<div
  className="makeStyles-container-1"
>
  <WithStyles(ForwardRef(MenuItem))
    className="makeStyles-menuItem-2"
    key="0"
  >
    <JobsLeftItem
      Icon={
        <Memo
          fontSize="inherit"
        />
      }
      key="0"
      title="Job Alerts"
      to="#"
    />
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    className="makeStyles-menuItem-2"
    key="1"
  >
    <JobsLeftItem
      Icon={
        <Memo
          fontSize="inherit"
        />
      }
      key="1"
      title="Skill Assessments"
      to="#"
    />
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    className="makeStyles-menuItem-2"
    key="2"
  >
    <JobsLeftItem
      Icon={
        <Memo
          fontSize="inherit"
        />
      }
      key="2"
      title="Interview Prep"
      to="#"
    />
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    className="makeStyles-menuItem-2"
    key="3"
  >
    <JobsLeftItem
      Icon={
        <Memo
          fontSize="inherit"
        />
      }
      key="3"
      title="Resume Builder"
      to="#"
    />
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    className="makeStyles-menuItem-2"
    key="4"
  >
    <JobsLeftItem
      Icon={
        <Memo
          fontSize="inherit"
        />
      }
      key="4"
      title="Application Settings"
      to="#"
    />
  </WithStyles(ForwardRef(MenuItem))>
</div>
`;
